checks/doks/dobs_pod_owner_test.go

```diff
@@ -0,0 +1,284 @@
+		{
+			name: "bare dobs pod referenced by pvc with default storage class",
+			objs: pvcDobs(""),
+			expected: []checks.Diagnostic{
+				{
+					Severity: checks.Warning,
+					Message:  "Pod referencing DOBS volumes must be owned by StatefulSet",
+					Kind:     checks.Pod,
+					Object:   &metav1.ObjectMeta{Name: "foo", Namespace: metav1.NamespaceDefault},
+					Owners:   nil,
+				},
+			},
+		},
+		{
+			name: "bare dobs pod referenced by csi",
+			objs: csiDobs(),
+			expected: []checks.Diagnostic{
+				{
+					Severity: checks.Warning,
+					Message:  "Pod referencing DOBS volumes must be owned by StatefulSet",
+					Kind:     checks.Pod,
+					Object:   &metav1.ObjectMeta{Name: "foo", Namespace: metav1.NamespaceDefault},
+					Owners:   nil,
+				},
+			},
+		},
+		{
+			name: "dobs pod owned by deployment",
+			objs: deployment(pvcDobs("")),
+			expected: []checks.Diagnostic{
+				{
+					Severity: checks.Warning,
+					Message:  "Pod referencing DOBS volumes must be owned by StatefulSet",
+					Kind:     checks.Pod,
+					Object: &metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: metav1.NamespaceDefault,
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion: "apps/v1",
+								Kind:       "Deployment",
+								Name:       "web-app",
+							},
+						},
+					},
+					Owners: []metav1.OwnerReference{
+						{
+							APIVersion: "apps/v1",
+							Kind:       "Deployment",
+							Name:       "web-app",
+						},
+					},
+				},
+			},
+		},
+		{
+			name:     "dobs pod owned by statefulset",
+			objs:     statefulSet(pvcDobs("")),
+			expected: nil,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			d, err := dobsPodOwner.Run(test.objs)
+			assert.NoError(t, err)
+			assert.ElementsMatch(t, test.expected, d)
+		})
+	}
+}
+
+func csiDobs() *kube.Objects {
+	objs := &kube.Objects{
+		Pods: &corev1.PodList{
+			Items: []corev1.Pod{
+				{
+					TypeMeta:   metav1.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+					ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: metav1.NamespaceDefault},
+					Spec: corev1.PodSpec{
+						Volumes: []corev1.Volume{
+							{
+								Name: "csi-do",
+								VolumeSource: corev1.VolumeSource{
+									CSI: &corev1.CSIVolumeSource{
+										Driver: DOCSIDriver,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	return objs
+}
+
+func statefulSet(objs *kube.Objects) *kube.Objects {
+	objs.Pods.Items[0].OwnerReferences = []metav1.OwnerReference{
+		{
+			APIVersion: "apps/v1",
+			Kind:       "StatefulSet",
+			Name:       "woot",
+		},
+	}
+	return objs
+}
+
+func deployment(objs *kube.Objects) *kube.Objects {
+	objs.Pods.Items[0].OwnerReferences = []metav1.OwnerReference{
+		{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+			Name:       "web-app",
+		},
+	}
+	return objs
+}
+
+func pvcDobs(storageClass string) *kube.Objects {
+	var sc *string
+	if storageClass == "" {
+		sc = nil
```

Comment:
This isn't needed since `sc` is declared as a pointer which defaults to the `nil` value.